memory/services/SqlServerMemoryStore.cs

```diff
@@ -246,34 +246,180 @@ public async Task ChangeConversationTopicAsync(Interaction changeTopic, Duration
+                return Task.CompletedTask;
+            });
     }
 
     public Task CommentOnMessageAsync(string userId, string activityId, string comment, CancellationToken cancellationToken = default)
     {
-        throw new HttpException(501, "not currently implemented");
+        return this.ExecuteWithRetryOnTransient(
+            async () =>
+            {
+                this.logger.LogDebug("attempting to update user {u} comment in the history database...", userId);
+                using var connection = this.GetConnection();
+                await connection.OpenAsync();
+                using var command = connection.CreateCommand();
+                using var transaction = await connection.BeginTransactionAsync(cancellationToken); // rollback is automatic during dispose
+                command.Transaction = (SqlTransaction)transaction;
+                command.CommandText = @"
+                    UPDATE [dbo].[History]
+                    SET [Comment] = @comment
+                    WHERE [UserId] = @userId AND [ActivityId] = @activityId AND [Role] = 'ASSISTANT';
+                ";
+                command.Parameters.AddWithValue("@comment", comment);
+                command.Parameters.AddWithValue("@userId", userId);
+                command.Parameters.AddWithValue("@activityId", activityId);
+
+                await command.ExecuteNonQueryAsync(cancellationToken);
+                await transaction.CommitAsync(cancellationToken);
+                this.logger.LogInformation("successfully update user {u} comment in the history database.", userId);
+            }, (ex, _) =>
+            {
+                this.logger.LogError(ex, "update comment for user {u} message raised the following SQL transient exception...", userId);
+                return Task.CompletedTask;
+            });
     }
 
-    public Task DeleteLastActivitiesAsync(string userId, int count = 1, CancellationToken cancellationToken = default)
+    public Task DeleteActivitiesAsync(string userId, int count = 1, CancellationToken cancellationToken = default)
     {
-        throw new HttpException(501, "not currently implemented");
+        return this.ExecuteWithRetryOnTransient(
+            async () =>
+            {
+                this.logger.LogDebug("attempting to delete user {u} message in the history database...", userId);
+                using var connection = this.GetConnection();
+                await connection.OpenAsync();
+                using var command = connection.CreateCommand();
+                using var transaction = await connection.BeginTransactionAsync(cancellationToken); // rollback is automatic during dispose
+                command.Transaction = (SqlTransaction)transaction;
+                command.CommandText = @"
+                    UPDATE [dbo].[History]
+                    SET [State] = 'DELETED'
```

Comment:
We need to clear the Message field as well.

Reply:
done